### ISSUE TITLE
Man page for iostat broken

### ISSUE THREAD

**Issue by OmnipotentEntity** [NixOS/nixpkgs#134805]
### Describe the bug
`man iostat` produces the following error:

```
/run/current-system/sw/bin/man: warning: /run/current-system/sw/share/man/man1/iostat.1.xz.gz: ignoring bogus filename
No manual entry for iostat
```

### Steps To Reproduce
1. Install the sysstat package (dil56c6nfhzwcy8jk9ckq08fxrkmxady-sysstat-12.4.3)
2. `man iostat`
3. Observe the error

### Expected behavior
The manual page for iostat(1)

### Notify maintainers
@edolstra 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.45, NixOS, 21.05.1176.0b8b127125e (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.12`
 - channels(root): `"nixos-21.05.1176.0b8b127125e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  sysstat
# a list of nixos modules affected by the problem
module:
```

**Comment by wentasah** [comment]
Yes, simple workaround is to add
```
dontGzipMan = true;
```
to `sysstat` Nix expression, but it would probably be more useful to update `compress-man-pages.sh` hook not to compress `.xz` files. I'll post a PR for that in a while.

**Comment by vcunat** [comment]
Fixed by PR #134881
(>= 21.11)
